components/insight/SRC/org/openmicroscopy/shoola/util/roi/io/ROIReader.java

```diff
@@ -342,58 +342,9 @@ private void setImage(Roi[] rois)
                     }
                 }
             } else if (r instanceof ShapeRoi) {
-                Roi[] subRois = ((ShapeRoi) r).getRois();
-                Roi shapeij;
-                for (int j = 0; j < subRois.length; j++) {
-                    shapeij = subRois[j];
-
-                    // Set ImagePlus reference in subROIs for the check in L216 to work
-                    ImagePlus imp = r.getImage();
-                    shapeij.setImage(imp);
-                    // Transfer correct ROI positions (according to IJ) from superROI
-                    int pos = r.getPosition();
-                    int c = r.getCPosition();
-                    int z = r.getZPosition();
-                    int t = r.getTPosition();
-                    if (imp.getNChannels() == 1 && imp.getNSlices() == 1) {
-                        shapeij.setPosition(pos);
-                    } else if (imp.getNChannels() == 1 &&
-                            imp.getNFrames() == 1) {
-                        shapeij.setPosition(pos);
-                    } else if (imp.getNSlices() == 1 &&
-                            imp.getNFrames() == 1) {
-                        shapeij.setPosition(pos);
-                    } else if (imp.isHyperStack()) {
-                        shapeij.setPosition(c, z, t);
-                    }
-                    type = shapeij.getTypeAsString();
```

Comment:
@stelfrich : The ROI type conversion logic seems to be removed. Currently you are just populating the subROI with the position alone. You will have to invoke the conversion logic, to actually create the corresponding omero subROI's. Due to this issue, the testing workflow is failing (silently) without creating any new ROI's on the imported image.